docs/docs/adding-pagination.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Adding pagination
+---
+
+A page displaying a list of content gets longer as the amount of content grows.
+Pagination is the technique of spreading that content across multiple pages.
+
+The goal is to create multiple pages (from a single [template](/docs/building-with-components/#page-template-components)) that show a limited number of items.
+
+Each page will [query GraphQL](/docs/querying-with-graphql/) for those specific items.
+
+The information needed to query for those specific items (a [limit](/docs/graphql-reference/#limit) and [skip](/docs/graphql-reference/#skip) value) will come from the [`context`](/docs/graphql-reference/#query-variables) that is added when [creating pages](/docs/creating-and-modifying-pages/#creating-pages-in-gatsby-nodejs) in `gatsby-node`
+
+### Example
+
+```js{20-25}:title=src/templates/blog-list-template.js
+import React from "react"
+import { graphql } from "gatsby"
+import Layout from "../components/layout"
+
+export default class BlogList extends React.Component {
+  render() {
+    const posts = this.props.data.allMarkdownRemark.edges
+    return (
+      <Layout>
+        {posts.map(({ node }) => {
+          const title = node.frontmatter.title || node.fields.slug
+          return <div key={node.fields.slug}>{title}</div>
+        })}
+      </Layout>
+    )
+  }
+}
+
+export const blogListQuery = graphql`
+  query blogListQuery($skip: Int!, $limit: Int!) {
+    allMarkdownRemark(
+      sort: { fields: [frontmatter___date], order: DESC }
+      limit: $limit
+      skip: $skip
+    ) {
+      edges {
+        node {
+          fields {
+            slug
+          }
+          frontmatter {
+            title
+          }
+        }
+      }
+    }
+  }
+`
+```
+
+```js{34-47}:title=gatsby-node.js
+const path = require("path")
+const { createFilePath } = require("gatsby-source-filesystem")
+
+exports.createPages = ({ graphql, actions }) => {
+  const { createPage } = actions
+
+  return new Promise((resolve, reject) => {
+    resolve(
+      graphql(
+        `
+          {
+            allMarkdownRemark(
+              sort: { fields: [frontmatter___date], order: DESC }
+              limit: 1000
+            ) {
+              edges {
+                node {
+                  fields {
+                    slug
+                  }
+                }
+              }
+            }
+          }
+        `
+      ).then(result => {
+        if (result.errors) {
+          reject(result.errors)
+        }
+
+        // ...
+
+        // Create blog-list pages
+        const posts = result.data.allMarkdownRemark.edges
+        const postsPerPage = 6
+        const numPages = Math.ceil(posts.length / postsPerPage)
+
+        Array.from({ length: numPages }).forEach((_, i) => {
+          createPage({
+            path: `/blog/${i + 1}`,
```

Comment:
Without making this too much more complex, I think it makes most sense to have `/blog/1` actually be `/blog`. Think that's doable? You can take a look at our [www](./www/gatsby-node.js) setup because we do something similar!